### ISSUE TITLE
Raising sv_fps breaks certain gameplay elements

### ISSUE THREAD

**Issue by Aciz** [etlegacy/etlegacy#1637]
Some gameplay elements are hardcoded to only function properly with `sv_fps 20` (50ms update rate on the server). Raising `sv_fps` to higher values, for example 40, breaks these things. So far the commonly documented issues are following:

* Covert ops steals disguises faster 
  * https://streamable.com/izgdtw

* Flamethrower hit detection range is increased (no visual change, only hit detection is affected)
  * https://streamable.com/nsxilx (`sv_fps 40`) & https://streamable.com/n218ao (`sv_fps 125`)

* `script_mover` has slight hitching/movement speed increase between spline points/path_corners. Might not be instantly visible here, but in the past I have experienced issues in ETJump with raising `sv_fps` for recording smooth demos in localhost, some jumps with `scirpt_movers` in them become impossible.
  * https://streamable.com/20w7k5 (slight mover hitching with railgun tugs)
  * https://streamable.com/j0y8es (impossible jump with high `sv_fps` values)

* Pausing a match with `/ref pause` causes a visual bug where match timer will keep counting down. This is a visual bug only - the timer corrects itself after the match is resumed. Happens only on certain `sv_fps` values.
  * https://streamable.com/ksjieu

If higher `sv_fps` values are desired to possibly improve hit detection/general snappiness of the game, these issues should be resolved and the corresponding actions made independent of the 50ms update rate from the server.

**Comment by rmarquis** [comment]
See related discussion on old ticket #285.

Especially, [this section](https://www.ra.is/unlagged/faq.html#ISFPSHT20SR) of the FAQ of unlagged:


>  Is setting sv_fps higher than 20 still recommended? (top)

> It's a somewhat common practice to raise the server framerate to 30 or 40 to make the gameplay “smoother.” What “smoother” means in this case is that Quake 3's built-in 50ms lag becomes 33ms or 25ms, and everyone sees things a bit more like how they happen on the server.
>
> Those are really the only two reasons to do it. Now, when Unlagged is running, full compensation or not, the built-in lag is taken care of completely. There goes one reason. It also does every hit test based on what players see rather than what's actually happening, so the second reason isn't any longer valid either.
>
> It does have some detrimental effects:
> 
> More incoming bandwidth is required of players
cl_timenudge can't extrapolate as much, meaning it's less effective (this becomes an issue when sv_fps is over 33)
Higher pings (it seems the client engine holds onto snapshots a little longer)
Other players may be “jerkier” since more samples mean less smoothing in general
The new skip correction becomes less effective
So no, I wouldn't advise higher server framerates. There aren't any advantages anymore, and there are plenty of disadvantages.

**Comment by Aciz** [comment]
Yeah I'm not entirely sure if making `sv_fps` higher is a good shortcut for better hitreg. From what I've read, proper antilag is gonna yield better results, but @jackeri said on discord that he would like to see these issues resolved so we could raise `sv_fps`, so I put all I have discovered so far in a single issue so it can be tracked more easily.

**Comment by ryzyk-krzysiek** [comment]
Writing down what I have found out and believe are the issues, so it doesn't get lost in discord.

- Covert ops steals disguises faster:

https://github.com/etlegacy/etlegacy/blob/a66af5eedc6f257340fbac04a2368bb7d02b7a98/src/game/g_cmds.c#L3772

This value gets incremented on every frame so the higher framerate the faster it gets to 250 where the stealing is finished. After discussing it with Aranud, the proposed solution could be to rework activate skill to work similar to engineer pliers.

- Flamethrower hit detection range is increased (no visual change, only hit detection is affected):

https://github.com/etlegacy/etlegacy/blob/0162ce044637c5ea0dccc76b6c874445b7219303/src/game/g_main.c#L5143

The flamechunk is run on every frame and the speed is adjusted which leads to higher speeds since it's done more often.

https://github.com/etlegacy/etlegacy/blob/abf688c362c29c786be4039c49c28033d6a0b842/src/game/g_missile.c#L993

It seems at some point it was limited to run only every 50ms which would fix the problem, but after discussing with Aranud the probably better solution would be to rework it to use `BG_EvaluateTrajectory` instead.

- Slight mover hitching with railgun tugs:

https://github.com/etlegacy/etlegacy/blob/00b35a7e14db6f0b73dd6fbf5fdfe7436a8ed776/src/game/g_script_actions.c#L758-L763

There is a prediction error/miss that causes bigger hitching the bigger the sv_fps setting, deleting above part fixes the issue. But there is still small hitching even after. I think the way the trains work is `trigger moving -> train moves -> train stops -> trigger moving ->  train moves -> train stops` and somehow on `sv_fps 20` and `sv_fps 40` the transition is much more seamless. But for example when train starts to rotate when it needs to turn it's much more seamless on `sv_fps 125` than on `sv_fps 20`. I didn't figure out how could it be fixed, but movers hitching might be only a problem on really high `sv_fps` like `sv_fps 100` which I think would be the only good setting after `40` but might be hard to achieve.

- Impossible jump with high sv_fps values:

https://github.com/etlegacy/etlegacy/blob/00b35a7e14db6f0b73dd6fbf5fdfe7436a8ed776/src/game/g_script_actions.c#L1696

The problem here is the `wait` command. This could be adjusted by changing `<` to `<=`. At this moment, `wait 150` command is actually `wait 200` this means that on `sv_fps 20` the pistons start moving at for example `level.time 200` while on `sv_fps 125` they would start moving on `level.time 152`. Changing the comparison operator would bring them much closer together but would mean that the map script would need to be updated from `wait 150` to `wait 200`. This also doesn't fix the problem if someone uses fe. a `wait 101` where with `sv_fps 20 vs sv_fps 125` would mean that `sv_fps 20` would be delayed once again by about 40ms. I don't think there is a real fix, but for a legacy mod gameplay and maps used there is probably no reason to not change it, as I don't think that difference which will come out because of it will have some big impact (but probably still it could be left as it is since all maps were designed with `<` operator).


- Pausing a match with /ref pause causes a visual bug:

https://github.com/etlegacy/etlegacy/blob/0162ce044637c5ea0dccc76b6c874445b7219303/src/game/g_main.c#L5243-L5248

Simply at some framerates `(level.time % 500) == 0` is never true, I think the solution may be in the `FIXME` comment.

- Not mentioned lagometer bug with average packet number received being incorrect:

https://github.com/etlegacy/etlegacy/blob/dfdb921c9c160c1c8c711a226fabbe5caee9a875/src/cgame/cg_local.h#L2258

I believe `LAG_SAMPLES` needs to be increased, the lagometer shows last 5 seconds and on higher `sv_fps` there is more snapshots and they are not getting stored. `sv_fps 20` 101 samples, `sv_fps 40` 201 samples.

**Comment by rmarquis** [comment]
Just as a general notice and as a follow up of #1408 : Keep in mind the section of the unlagged FAQ posted above.
If interpolation of antilag is properly fixed, raising `sv_fps` might not yield better results (and could actually be detrimental).

Proper testing needed.

**Comment by ryzyk-krzysiek** [comment]
`WolfRevivePushEnt` is affected by `sv_fps` making the push stronger the higher the fps, but it makes getting stuck inside of player after revive on purpose impossible.

https://github.com/etlegacy/etlegacy/blob/28a4ae45d6acf48743e32c775b6ed21d22066f9f/src/game/g_active.c#L1963-L1989
